src/plugins/intel_cpu/src/nodes/lora.cpp

```diff
@@ -86,24 +88,39 @@ void LoRA::selectOptimalPrimitiveDescriptor() {
     selectPrimitiveDescriptorByIndex(0);
 }
 
-// @todo add ascii diagram for memory mapping / reuse
+int LoRA::registerToAllocationContext(int offset, AllocationContext& context) {
+    for (size_t i = 0; i < getOriginalInputsNumber(); i++) {
+        auto parentEdge = getParentEdgeAt(i);
+        auto inputEdges = m_graph.GetInputNodesMap().at(i)->getChildEdgesAtPort(0);
+        for (const auto& inputEdge : inputEdges) {
+            OPENVINO_ASSERT(inputEdge->getStatus() == Edge::Status::Uninitialized,
+                            "Expected Uninitialized Edge instead of: ", static_cast<int>(inputEdge->getStatus()));
+            inputEdge->sharedMemFrom(parentEdge);
+        }
+    }
+
+    for (size_t i = 0; i < getOriginalOutputsNumber(); i++) {
+        auto childEdge = getChildEdgeAt(i);
+        auto outputEdge = m_graph.GetOutputNodesMap().at(i)->getParentEdgeAt(0);
+        outputEdge->sharedMemFrom(childEdge);
+    }
+
+    return m_graph.RegisterToAllocationContext(offset, context);
+}
+
 void LoRA::createPrimitive() {
     CPU_NODE_ASSERT(getOriginalInputsNumber() == m_graph.GetInputNodesMap().size(),
                     "Number of node inputs must be equal the number of inner graph's inputs");
-
-    std::vector<MemoryPtr> inputMemory;
+    // Workaround to avoid making LoRa node always executable (isExecutable() = true)
+    // This way we update subgraph's input memory without performing an actual Infer() call
     for (size_t i = 0; i < getOriginalInputsNumber(); i++) {
-        auto srcEdgeMem = getSrcMemoryAtPort(i);
-        auto mem = std::make_shared<Memory>(getEngine(), srcEdgeMem->getDescPtr(), srcEdgeMem->getMemoryBlock());
+        const auto& subgraphInputNode = m_graph.GetInputNodesMap().at(i);
+        const auto& subgraphInputMemory = subgraphInputNode->getDstMemoryAtPort(0);
+        auto mem = std::make_shared<Memory>(getEngine(), subgraphInputMemory->getDescPtr(), subgraphInputMemory->getMemoryBlock());
         subgraphMemoryPtrs.push_back(mem);
```

Comment:
I thought this is one of the ideas behind the workaround - to not use original memory.
Am I missing something?
Or you mean it is not applicable anymore based on this PR changes?

Reply:
The thing is, on master `mem` is used inside the subgraph: we create it first using the the memory block from the parent graph, store it in `subgraphMemoryPtrs` and **_then we put it to the subgraph via the `Activate` call._** But in the PR we bind the edges, not the memory objects, so that they share the common memory block, but have different memory objects automatically (we don't need to create new memory objects for the subgraph input edges and put it to the subgraph via `Activate`). So the `mem` is simply stored in `subgraphMemoryPtrs` and share the same memory block, but it's not the same memory object from the subgraph. However, we still have to transfer the shape, as the internal subgraph input memory object is different from what is used in the parent graph. Someone needs to transfer the shape. Thus. Instead of creating the separate `mem` objects, we need to store the input memory objects of the internal subgraph to access them efficiently and update the shapes.